Ad9545 add fast acquisition

tachicialex opened the pull request:
In fast acquisition the DPLL automatically reduces
the loop bandwidth by successive factors of 2 while the loop is
acquiring.

This mechanism drastically reduces DPLL locking time
for low loop bandwidth applications like 1 pps signals.

nunojsa commented on drivers/clk/adi/clk-ad9545.c:
```diff
@@ -1749,6 +1786,42 @@ static int ad9545_plls_setup(struct ad9545_state *st)
+				if (tmp == ad9545_fast_acq_excess_bw_map[k])
+					break;
+
+			if (k < ARRAY_SIZE(ad9545_fast_acq_excess_bw_map)) {
```
What if `k == ARRAY_SIZE(ad9545_fast_acq_excess_bw_map)`? This means that the dts property given by the user is wrong and we are silently ignoring it, right? Maybe this would make more sense?

```
if (k >= ARRAY_SIZE(ad9545_fast_acq_excess_bw_map)`) {
        dev_err(...);
        return -EINVAL;
}
```
If the user is using this and for some reason just got it wrong, we should be verbose about it...

nunojsa commented on drivers/clk/adi/clk-ad9545.c:
```diff
@@ -1749,6 +1786,42 @@ static int ad9545_plls_setup(struct ad9545_state *st)
+				if (tmp == ad9545_fast_acq_timeout_map[k])
+					break;
+
+			reg |= k << 4;
```
ditto... maybe check for a valid index? Another alternative is to just check for valid properties when parsing them (maybe it even makes more sense) and then you don't really need to this here. The downside is that you will have to save them in a way ready to be written to the device...

Anyways, that's up to you. I'm fine either way...

nunojsa reviewed (commented):
LGTM, just some minor comments...

tachicialex commented:
Changelog v1 -> v2:
- moved fast acquisition setup from ad9545_plls_setup
to ad9545_pll_fast_acq_setup
- added error messages in case of wrong values

nunojsa commented on drivers/clk/adi/clk-ad9545.c:
```diff
@@ -1684,6 +1719,63 @@ static const struct clk_ops ad9545_pll_clk_ops = {
+			pll->address, profile);
+		return -EINVAL;
+	}
+
```
nit: You could do as you did in the following cases. Just check for the invalid case and return. Hence, no need for `if...else`. Just:

```
if (i == ARRAY_SIZE(ad9545_fast_acq_excess_bw_map)) {
      dev_err();
      return -EINVAL;
}

ret = regmap_write();
....
```

nunojsa reviewed (approved):
Just a minor nit: I will leave this a bit more time in case you want to change it. If not, will merge it Monday
